src/components/cubecomps/FormCubeArrays.jsx

```diff
@@ -1,257 +1,26 @@
-        uvs.push(...onefaceuv);
-        dbstr = dbstr + "right\n";
-        faceindexmap[facemapcount] = makeFaceIndexMapObject(cen, nx + 1, ny, nz);
-        facemapcount += 2;
-      }
-      // //top
-      if (!blocks[makeKey(nx, ny + t2, nz)]) {
-        vertices.push(...c[4], ...c[8], ...c[3]);
-        vertices.push(...c[3], ...c[8], ...c[7]);
-        normals.push(0, 1, 0, 0, 1, 0, 0, 1, 0, 0, 1, 0, 0, 1, 0, 0, 1, 0);
-        uvs.push(...onefaceuv);
-        dbstr = dbstr + "top\n";
-        faceindexmap[facemapcount] = makeFaceIndexMapObject(cen, nx, ny + 1, nz);
-        facemapcount += 2;
-      }
-      // //bot
-      if (!blocks[makeKey(nx, ny - t2, nz)]) {
-        vertices.push(...c[5], ...c[1], ...c[6]);
-        vertices.push(...c[6], ...c[1], ...c[2]);
-        normals.push(0, 1, 0, 0, 1, 0, 0, 1, 0, 0, 1, 0, 0, 1, 0, 0, 1, 0);
-        uvs.push(...onefaceuv);
-        dbstr = dbstr + "bot\n";
-        faceindexmap[facemapcount] = makeFaceIndexMapObject(cen, nx, ny - 1, nz);
-        facemapcount += 2;
-      }
-
-      blocks[cen].showface = showfaces;
-    });
-
-    cubeFaceIndexesREF.current = faceindexmap;
-
-    return [vertices, uvs, normals];
-  }
-
-  function makeKey(x, y, z) {
-    return x + "." + y + "." + z;
-  }
-
-  function updateDataContainer(blocks) {
-    let t = 0.5;
-    let allkeys = Object.keys(blocks);
-    let [vertices, uvs, normals, faceTextures] = genFaceArrays(t, allkeys, blocks);
-    vertices = new Float32Array(vertices);
-    uvs = new Float32Array(uvs);
-    normals = new Float32Array(normals);
-
-    let newAC = [];
-
-    newAC.push({
-      vertices,
-      normals,
-      uvs,
-      allkeys,
-    });
-    setDataCont(newAC);
-  }
-
-  function clickCubeFace(e) {
-    // e.preventDefault()
-    e.stopPropagation(); //click cannot be passed through cube face
-    var raycaster = new THREE.Raycaster();
-    var mouse = new THREE.Vector2();
-    mouse.x = (e.clientX / window.innerWidth) * 2 - 1;
-    mouse.y = -(e.clientY / window.innerHeight) * 2 + 1;
-    raycaster.setFromCamera(mouse, camera);
-
-    let intersect = raycaster.intersectObjects(scene.children);
-
-    intersect = intersect.filter((inter) => {
-      return inter.object.name == "cubesMesh2";
-    });
-
-    if (intersect.length > 0) {
-      let currBlocks = getAllBlocks();
-
-      let f_Index = intersect[0].faceIndex;
-      f_Index = f_Index - (f_Index % 2);
-
-      let currTexture = activeTextureREF.current;
-
-      if (e.which === 1) {
-        let newblock = cubeFaceIndexesREF.current[f_Index].add;
-        currBlocks[newblock.key] = { pos: newblock.pos, texture: currTexture };
-        updateAllBlocks(currBlocks);
-      }
-
-      if (e.which === 3) {
-        let remove = cubeFaceIndexesREF.current[f_Index].remove;
-        delete currBlocks[remove];
-        updateAllBlocks(currBlocks);
+export const FormCubeArrays = ({chunkNum, clickCubeFace, chunkProps }) => {
```

Comment:
Just to make sure I'm understanding what a cube array is, is a cube array essentially all the block data in one chunk?